packages/vite-plugin-svelte/src/utils/id.js

```diff
@@ -179,10 +180,15 @@ function buildFilter(include, exclude, extensions) {
  */
 function buildModuleFilter(include, exclude, infixes, extensions) {
 	const rollupFilter = createFilter(include, exclude);
-	return (filename) =>
-		rollupFilter(filename) &&
-		infixes.some((infix) => filename.includes(infix)) &&
-		extensions.some((ext) => filename.endsWith(ext));
+	return (filename) => {
+		const basename = path.basename(filename);
+
+		return (
+			rollupFilter(filename) &&
+			infixes.some((infix) => basename.includes(infix)) &&
+			extensions.some((ext) => filename.endsWith(ext))
```

Comment:
Isn't it better to also use the basename here? Where would that return a false negative?
```suggestion
			extensions.some((ext) => basename.endsWith(ext))
```

Reply:
Identical result, which is why I didn't bother changing it. But certainly no harm in it